tests/test_plugin.py

```diff
@@ -140,3 +143,38 @@ def test_account(fxa_urls):
     """)
     result = testdir.runpytest()
     result.assert_outcomes(passed=1)
+
+
+def test_cleanup_after_failed_verification(mocker, testdir):
+    def mock_headers(arg1, arg2):
+        """Mock x-verify-code header to return an invalid code."""
+        _ = {'headers': {}}
+        _['headers']['x-verify-code'] = '{}'.format(
+            binascii.b2a_hex(os.urandom(16)).decode()
+        )
+        return _
+
+    mocker.patch.object(TestEmailAccount, 'wait_for_email', mock_headers)
+    testdir.makepyfile("""
+        import pytest
+
+        def test_login(fxa_account, fxa_client):
+            assert fxa_client.login(fxa_account.email, fxa_account.password)
+    """)
+    result = testdir.runpytest()
+    result.assert_outcomes(error=1)
+    assert 'ClientError: Invalid verification code' in result.stdout.str()
```

Comment:
This looks like it would be a good test for PyFxA itself. For this plugin, I would suggest mocking `verify_email_code` so that it raises an exception, and adding a spy to `fxa_cleanup` so that we can verify that it is called.

Reply:
By mocking ```wait_for_email``` I can make ```verify_email_code``` return an exception. This is what I did here. ```verify_email_code``` tries to verify a code but it is incorrect.

Reply:
By mocking `wait_for_email` you're relying on the implementation of `verify_email_code` to raise an exception, and you're checking the specific exception. This is why it's a good test for PyFxA, but in our case we just want an exception and to verify in that scenario that the cleanup method is called. It doesn't really matter which method raises the exception, and it shouldn't matter what exception is raised.